File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/InputApiKeyAuth.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

namespace Microsoft.Generator.CSharp.Input
{
    public class InputApiKeyAuth
    {
        public InputApiKeyAuth(string name, string? prefix)
        {
            Name = name;
            Prefix = prefix;
        }

        public string Name { get; }
        public string? Prefix { get; }
    }
}

File: packages/http-client-csharp/emitter/src/type/input-api-key-auth.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

export interface InputApiKeyAuth {
  name: string;
  in: ApiKeyLocation;
  prefix?: string;
}

export type ApiKeyLocation = "header"; // | "query" | "cookie"; // we do not support query or cookie yet
